src/spn/algorithms/Inference.py

```diff
@@ -39,6 +39,21 @@ def prod_likelihood(node, children, data=None, dtype=np.float64):
     assert llchildren.dtype == dtype
     return np.prod(llchildren, axis=1).reshape(-1, 1)
 
+#added
+def max_log_likelihood(node, children, data=None, dtype=np.float64):
+    llchildren = np.concatenate(children, axis=1)
+    assert llchildren.dtype == dtype
```

Comment:
I'm a little confused as to the necessity of this assert -- isn't it just better to explicitly type caste `llchildren` to the type you'd like it to be? Is there a reason you have chosen to do this?

Reply:
I used it to be consistent with other methods already implemented for sum and prod likelihoods. I kept the template and just changed the return statements.

Reply:
Okay, I see, I think this should be changed everywhere then, since the assert is sort of meaningless.